src/pybind/mgr/dashboard/frontend/src/app/shared/api/upgrade.service.ts

```diff
@@ -0,0 +1,48 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { ApiClient } from './api-client';
+import { map } from 'rxjs/operators';
+import { SummaryService } from '../services/summary.service';
+import { UpgradeInfoInterface } from '../models/upgrade.interface';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class UpgradeService extends ApiClient {
+  baseURL = 'api/cluster/upgrade';
+  versionNumber: string;
+
+  constructor(private http: HttpClient, private summaryService: SummaryService) {
+    super();
+  }
+
+  list() {
+    return this.http.get(this.baseURL).pipe(
+      map((resp: UpgradeInfoInterface) => {
+        return this.versionAvailableForUpgrades(resp);
+      })
+    );
+  }
+
+  versionAvailableForUpgrades(upgradeInfo: UpgradeInfoInterface): UpgradeInfoInterface {
+    this.summaryService.subscribe((summary) => {
+      const version = summary.version.replace('ceph version ', '').split('-');
```

Comment:
Maybe this code snippet can be made simpler by directly filtering `upgradeInfo.versions`?
```suggestion
const versionNumber = summary.version.replace('ceph version ', '').split('-')[0];
const upgradableVersions = upgradeInfo.versions.filter(targetVersion => {
  const cVersion = versionNumber.split('.');
  const tVersion = targetVersion.split('.');

  return cVersion[0] === tVersion[0] && (cVersion[1] < tVersion[1] || cVersion[2] < tVersion[2]);
});

upgradeInfo.versions = upgradableVersions;
return upgradeInfo; 
```

Reply:
thanks @avanthakkar updated!